### PR TITLE
Add nested pipeline example:

### PR DESCRIPTION
**Pull request by kevinyu98**
**Which issue is resolved by this Pull Request:** 
Resolves #

**Description of your changes:**

- add nested pipeline example

**Environment tested:**

* Python Version (use `python --version`):3.6.4
* Tekton Version (use `tkn version`):0.11.3
* Kubernetes Version (use `kubectl version`):1.16
* OS (e.g. from `/etc/os-release`):

### CODE REVIEW

**Comment by kevinyu98** [comment]
@Tomcli

File: samples/nested-pipeline/README.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# Nested Pipeline example
+This pipeline example demonstrates that within a pipeline, you can call another pipeline during execution, also known as nested pipeline. This example is originated from the kubeflow pipeline's [compose.py](https://github.com/kubeflow/pipelines/blob/master/sdk/python/tests/compiler/testdata/compose.py) example that is running with the KFP API. We have modified this notebook to run with the Tekton compiler and submit the pipeline directly to the Tekton engine.
```
[the remark this answers is not in the document]
Comment (kevinyu98) [reply samples/nested-pipeline/README.md:2]: ok

File: samples/nested-pipeline/README.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+This example contains two pipelines, `save-most-frequent` and `download-and-save-most-frequent`. `save-most-frequent` will take two inputs, one is a message string which you want to find the most frequent word in it, the other is the location/path you want to save in GCS, if you don't have GCS setup, this parameter will be ignored, and will not save. This pipeline will get the input string from user, then find the most frequent used word in the string, print and save the word. `download-and-save-most-frequent` will take two inputs, one is an URL with plain text, the second one is the location/path which you want to save the result in GCS. This pipeline will first download the text from the input URL, then call `save-most-frequent` pipeline.
```
[the remark this answers is not in the document]
Comment (kevinyu98) [reply samples/nested-pipeline/README.md:4]: ok

File: samples/nested-pipeline/compose.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
+# Copyright 2018 Google LLC
```
[the remark this answers is not in the document]
Comment (kevinyu98) [reply samples/nested-pipeline/compose.py:1]: done

File: samples/nested-pipeline/compose.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        name=name,
+        image='google/cloud-sdk',
+        command=['sh', '-c'],
+        arguments=['echo \'%s\' | tee /tmp/results.txt | gsutil cp /tmp/results.txt \'%s\'' % (message, output_path)])
+
+@dsl.pipeline(
+  name='Save Most Frequent',
+  description='Get Most Frequent Word and Save to GCS'
+)
+def save_most_frequent_word(message: dsl.PipelineParam, outputpath: dsl.PipelineParam):
```
[the remark this answers is not in the document]
Comment (kevinyu98) [reply samples/nested-pipeline/compose.py:62]: ok

File: samples/nested-pipeline/compose.py [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        arguments=['gsutil cat %s | tee /tmp/results.txt' % url],
+        file_outputs={'downloaded': '/tmp/results.txt'}
+    )
+
+
+@dsl.pipeline(
+  name='Download and Save Most Frequent',
+  description='Download and Get Most Frequent Word and Save to GCS'
+)
+def download_save_most_frequent_word(url: str, outputpath: str):
```
[the remark this answers is not in the document]
Comment (kevinyu98) [reply samples/nested-pipeline/compose.py:98]: ok

File: samples/nested-pipeline/compose.py [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+  downloader = DownloadMessageOp('download', url)
+  save_most_frequent_word(downloader.output, outputpath)
```
[the remark this answers is not in the document]
Comment (kevinyu98) [reply samples/nested-pipeline/compose.py:100]: ok

File: samples/nested-pipeline/compose.py [hunk 7]
```diff
@@ -95,6 +97,12 @@ def __init__(self, name, url):
...
-def download_save_most_frequent_word(url: str, outputpath: str):
+def download_save_most_frequent_word(
+                                url="gs://ml-pipeline-playground/shakespeare1.txt", 
+                                outputpath="res.txt"):
   downloader = DownloadMessageOp('download', url)
   save_most_frequent_word(downloader.output, outputpath)
+
+if __name__ == '__main__':
+    from kfp_tekton.compiler import TektonCompiler
+    TektonCompiler().compile(download_save_most_frequent_word, __file__.replace('.py', '.yaml'))
```
Comment (Tomcli) [inline samples/nested-pipeline/compose.py:108]: nit: add newline to the end of file.

File: samples/nested-pipeline/compose.py [hunk 8]
```diff
@@ -95,6 +97,12 @@ def __init__(self, name, url):
   name='Download and Save Most Frequent',
   description='Download and Get Most Frequent Word and Save to GCS'
 )
-def download_save_most_frequent_word(url: str, outputpath: str):
+def download_save_most_frequent_word(
+                                url="gs://ml-pipeline-playground/shakespeare1.txt", 
+                                outputpath="res.txt"):
```
Comment (Tomcli) [inline samples/nested-pipeline/compose.py:102]: we should keep the type checking while having default variables. e.g.
```suggestion
                                outputpath: str ="res.txt"):
```

File: samples/nested-pipeline/compose.py [hunk 9]
```diff
@@ -59,7 +59,9 @@ def __init__(self, name, message, output_path):
   name='Save Most Frequent',
   description='Get Most Frequent Word and Save to GCS'
 )
-def save_most_frequent_word(message: dsl.PipelineParam, outputpath: dsl.PipelineParam):
+def save_most_frequent_word(
+              message="hello world hello", 
+              outputpath="result.txt"):
```
Comment (Tomcli) [inline samples/nested-pipeline/compose.py:64]: same as above

File: samples/nested-pipeline/compose.py (hunk 7 above)
Comment (kevinyu98) [reply samples/nested-pipeline/compose.py:108]: ok

File: samples/nested-pipeline/compose.py (hunk 8 above)
Comment (kevinyu98) [reply samples/nested-pipeline/compose.py:102]: thanks.

File: samples/nested-pipeline/compose.py (hunk 9 above)
Comment (kevinyu98) [reply samples/nested-pipeline/compose.py:64]: done.

**Comment by Tomcli** [comment]
Thanks @kevinyu98, it looks good to me. Since Animesh added a new Readme for samples, can you hyperlink this example to the list here as well?
https://github.com/kubeflow/kfp-tekton/blob/master/samples/README.md

**Comment by kevinyu98** [comment]
@Tomcli sure, thanks.

File: samples/nested-pipeline/README.md [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+```
+Waiting for logs to be available...
+[get-frequent : get-frequent] hello
+
+[save : save] {{inputs.parameters.get-frequent-word}}
+
+```
+To tryout the nested pipeline `download-and-save-most-frequent`:
+```
+dsl-compile-tekton --py compose.py --output pipeline.yaml --function download_save_most_frequent_word
```
Comment (Tomcli) [inline samples/nested-pipeline/README.md:34]:
```suggestion
dsl-compile-tekton --py compose.py --output pipeline.yaml --function download_save_most_frequent_word
kubectl apply -f pipeline.yaml
```

File: samples/nested-pipeline/README.md [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+## Prerequisites
+- Install [Kubeflow 1.0.2+](https://www.kubeflow.org/docs/started/getting-started/) and connect the cluster to the current shell with `kubectl`
+- Install [kfp-tekton](/sdk/README.md#steps) SDK
+
+## Instructions
+
+Once you have completed all the prerequisites for this example, you can first try out the `save-most-frequent` pipeline
+```
+dsl-compile-tekton --py compose.py --output pipeline.yaml --function save_most_frequent_word
+
```
Comment (Tomcli) [inline samples/nested-pipeline/README.md:15]:
```suggestion
kubectl apply -f pipeline.yaml
```

File: samples/nested-pipeline/README.md (hunk 10 above)
Comment (kevinyu98) [reply samples/nested-pipeline/README.md:34]: thanks, done.

File: samples/nested-pipeline/README.md (hunk 11 above)
Comment (kevinyu98) [reply samples/nested-pipeline/README.md]: thanks, done.

**Pull request merged**